### PR TITLE
fix(augmentation): prevent double application of probability p in RandomMixUpV2 and RandomCutMixV2 (#4649)

### LINKED ISSUE #4649: RandomMixUpV2 / RandomCutMixV2 apply `p` twice: a batch-wide gate and a per-row Bernoulli, so rows are mixed at rate p²
## Summary
`RandomMixUpV2(p=...)` and `RandomCutMixV2(p=...)` map `p` to the base **batch-wide** gate (`p=1.0, p_batch=p`), and then their generators draw a second, **per-row** `Bernoulli(p)` that zeroes `mixup_lambdas` / the cut size. A row is therefore mixed with probability `p²`, not `p`, and inside a "selected" batch individual rows are silently left unmixed.

## Reproduction
```python
import torch, kornia.augmentation as K
x = torch.rand(4, 1, 8, 8); sel = rows = mixed = 0
for s in range(400):
    torch.manual_seed(s); a = K.RandomMixUpV2(p=.5); a(x)
    bp = a._params["batch_prob"]; assert bp.unique().numel() == 1   # batch-wide gate
    if bp[0] > 0:
        sel += 1; rows += 4; mixed += int((a._params["mixup_lambdas"] > 0).sum())
print(sel, mixed, rows)   # 195 selected batches; 405 / 780 rows actually mixed
```
Same loop with `RandomCutMixV2(p=.5, use_correct_lambda=True)`: 295 / 780 rows changed inside selected batches. Per-row rate ≈ 0.5 · 0.5 = 0.25 rather than the requested 0.5. At `p=1` both draws are always true, which is why no test sees it.

Source: `random_generator/_2d/mixup.py` (`mixup_lambdas = mixup_lambdas * batch_probs`) and the matching `cutmix.py` generator, on top of `p_batch=p` in the class constructors.

## Expected
`p` applied once. Either the batch gate or the per-row draw, not both.

Measured on `main` @ c643312a3 merged with #4634 (docs-only), torch 2.14.0, CPU, macOS arm64. Found while reviewing #4634 (batch 6d of #4407).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

### PR DESCRIPTION
**Pull request by Prakhar00001**
## Description
Fixes #4649

`RandomMixUpV2` and `RandomCutMixV2` mapped `p` to `p_batch=p` in `super().__init__()` while simultaneously passing `p=p` into their underlying generators (`MixupGenerator` and `CutmixGenerator`). Because the generator independently sampled a per-row `Bernoulli(p)` mask, individual rows inside an already-selected batch were zeroed out at rate `1 - p`. Consequently, rows were mixed at an effective rate of $p^2$ instead of $p$.

This PR sets `p=1.0` in the internal parameter generators so that when a batch is gated on via `p_batch=p`, all rows in that batch are mixed.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update

## Changes Made
- `kornia/augmentation/_2d/mix/mixup.py`: Set `p=1.0` when instantiating `MixupGenerator`.
- `kornia/augmentation/_2d/mix/cutmix.py`: Set `p=1.0` when instantiating `CutmixGenerator`.
- `tests/augmentation/test_augmentation_mix.py`: Added regression unit tests `test_mixup_prob_single_gate` and `test_cutmix_prob_single_gate` checking that `_param_generator.p == 1.0` and that all rows in an active batch are mixed.
- `changelog.d/4649.fixed.md`: Added changelog fragment.

## How Has This Been Tested?
- Verified with the empirical reproduction script from #4649: batch-selected rows now maintain a 100% mix rate inside active batches, yielding an overall mix rate equal to $p$.
- Ran regression tests locally:
  ```bash
  python -m pytest tests/augmentation/test_augmentation_mix.py -k "test_mixup_prob_single_gate or test_cutmix_prob_single_gate" -v

### CODE REVIEW

**Review by ducha-aiki — APPROVED** [review]
Applying `p` solely at the batch gate removes the second independent probability draw in MixUp/CutMix and matches the documented contract. Both new focused regressions pass, and the current remote CI matrix is green. No blocking correctness findings.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).
